### PR TITLE
[CH] Issue 5018

### PR DESCRIPTION
**Pull request by binmahone**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#5018)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by binmahone** [comment]
This PR is is blocked by https://github.com/apache/incubator-gluten/pull/4676, needs its bug fix

File: backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/datasources/v1/CHMergeTreeWriterInjects.scala [hunk 1]
```diff
@@ -83,11 +86,14 @@ class CHMergeTreeWriterInjects extends GlutenFormatWriterInjectsBase {
...
       lowCardKeyOption,
+      minmaxIndexKeyOption,
+      bfIndexKeyOption,
+      setIndexKeyOption,
       primaryKeyOption,
       partitionColumns,
       ConverterUtils.convertNamedStructJson(tableSchema),
       clickhouseTableConfigs,
-      dataSchema
+      tableSchema.toAttributes // use table schema instead of data schema
```
Comment (binmahone) [inline backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/datasources/v1/CHMergeTreeWriterInjects.scala:96]: @zzcclp , pay attention to this

Comment (zzcclp) [reply backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/datasources/v1/CHMergeTreeWriterInjects.scala:96]: does it make sure that the `dataSchema` is the same as the tableSchema, especially the nullable property?
If yes, can remove the dataSchema parameter.

Comment (binmahone) [reply backends-clickhouse/src/main/scala/org/apache/spark/sql/execution/datasources/v1/CHMergeTreeWriterInjects.scala:95]: yes, done

**Review by zzcclp — APPROVED** [review]
LGTM.
Please update the PR title.

**Pull request merged**